azurerm/resource_arm_redis_cache.go

```diff
@@ -186,6 +186,12 @@ func resourceArmRedisCache() *schema.Resource {
 							Optional:  true,
 							Sensitive: true,
 						},
+						"authnotrequired": {
```

Comment:
I think it'd be clearer to invert this and make this a bool (with a default value of 'true'); and then handle the parsing to/from `yes/no` internally:

```suggestion
						"enable_authentication": {
```

WDYT?

Reply:
Thanks Tom, `enable_authentication` is definitely a better name as opposed to `authnotrequired`. I have made the necessary changes to reflect this.